Mobile deployment docs overhaul (#4606)
mattermost/docs: source/mobile/use-prebuilt-mobile-apps.rst

Using Mattermost's Pre-Built Mobile Apps
========================================

We strongly recommend using Mattermost’s pre-built Mobile App. This approach is recommended for all Mattermost customers.

Deployment Options
------------------

> When you decide to use the pre-built Mattermost Mobile apps, you have two ways to deploy them: through public app stores, or through an Enterprise Mobile Management (EMM) provider.

Reviewer: ```suggestion
When you decide to use the pre-built Mattermost Mobile apps, you have two ways to distribute them: ask your users to download it through public app stores, or through an Enterprise Mobile Management (EMM) provider.
```

Using Mattermost's Pre-Built Mobile Apps
========================================

> We strongly recommend using Mattermost’s pre-built Mobile app. This approach is recommended for Mattermost Cloud and Enterprise Edition customers.

Reviewer 2: ```suggestion
We strongly recommend using Mattermost’s pre-built Mobile App. This approach is recommended for Mattermost Cloud and Enterprise Edition customers.
```

Deployment Options
------------------

When you decide to use the pre-built Mattermost Mobile apps, you have two ways to deploy them: through public app stores, or through an Enterprise Mobile Management (EMM) provider.
  
Using Public App Stores
~~~~~~~~~~~~~~~~~~~~~~~

Your users can download the Mattermost Mobile app as an `iOS app <https://about.mattermost.com/mattermost-ios-app/>`__ or an `Android app <https://about.mattermost.com/mattermost-android-app/>`__ from a public app store. When users launch the Mattermost Mobile app, they must enter the address of your Mattermost server to connect their Mobile app to the server.

Using an EMM Provider
~~~~~~~~~~~~~~~~~~~~~

See our `Deploying Using an EMM Provider <https://docs.mattermost.com/mobile/deploy-mobile-apps-using-emm-provider.html>`__ documentation to learn more about deploying Mattermost Mobile apps through an EMM provider.

> Setting Up Push Notifications

Reviewer 3: Also note that TPNS is automatically configured out-of-the-box with new servers (so nothing actually has to be done to enable it)

> Setting Up Push Notifications

Author: I was considering this very idea. Thanks! Will move this content to its own section so that it can be easily linked to from either flow.

-----------------------------

Mattermost hosts a push proxy option available for Mattermost Cloud and Enterprise Edition deployments. The HPNS offers:

- An explicit `privacy policy <https://mattermost.com/data-processing-addendum/>`__ for the contents of unencrypted messages.

> - Encrypted TLS connections

Reviewer 2: ```suggestion
- Encrypted TLS connections:
```

    - Between HPNS and Apple Push Notification Services
    - HPNS and Google’s Firebase Cloud Messaging service
    - HPNS and your Mattermost Server
- Production-level uptime expectations.

> - Compatibility with EMM Providers.

Author: @esethna - are you able to shed some light on this one?

> - Compatibility with EMM Providers.

Reviewer 3: Not sure what that means tbh

Configuring your Mattermost instance to use the Mattermost HPNS is a single, one-time step. Learn more about securing and configuring the `HPNS <https://docs.mattermost.com/mobile/mobile-hpns.html>`__ in our product documentation. 

ID-Only Push Notifications
~~~~~~~~~~~~~~~~~~~~~~~~~~

> Mattermost Cloud and Enterprise customers can limit the data sent to the HPNS through a configuration setting. 

Reviewer 2: ```suggestion
Mattermost Cloud and Enterprise Edition customers can limit the data sent to the HPNS through a configuration setting. 
```

> When enabled, a message containing only an ID is transmitted. Once the mobile client receives this ID, the message contents are loaded from the server, and are never transmitted through Apple Push Notification Service (APNS) or Firebase Cloud Messaging (FCM). See our `Configuration Settings <https://docs.mattermost.com/administration/config-settings.html#push-notification-contents>`__ documentation to learn more about ID-only push notifications.

Author: Thanks! I've added this to the documentation.

Test Push Notifications Service (TPNS)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> After setting up push notifications, we strongly recommend that you test push notifications to ensure they are working. 

Reviewer 2: ```suggestion
After setting up push notifications, we strongly recommend that you test push notifications to ensure they're working.
```

> After setting up push notifications, we strongly recommend that you test push notifications to ensure they're working.

Reviewer 3: Agree, we want them to "test" notifications but it doesn't need to be with TPNS. It could be with HPNS if they are on EE

> Mattermost offers a free basic hosted service to test push notifications for self-managed deployments. The TPNS isn’t recommended for use in Production environments, and doesn’t offer production-level update service level agreements (SLAs). To use the Mattermost TPNS:

Reviewer 2: ```suggestion
Mattermost offers a free basic hosted service to test push notifications for self-managed deployments. The TPNS isn’t recommended for use in production environments, and doesn’t offer production-level update service level agreements (SLAs). To use the Mattermost TPNS:
```

> 1. In System Console, go to **Environment > Push Notification Server > Enable Push Notifications**, then select **Use TPNS connection to send notifications to iOS and Android apps**.

Reviewer 2: ```suggestion
1. Go to **System Console > Environment > Push Notification Server > Enable Push Notifications**, then select **Use TPNS connection to send notifications to iOS and Android apps**.
```

> 2. Specify the URL of the **Push Notification Server** based on your Mattermost edition.

Author: HPNS comes set by default out of the box, so it's necessary to switch to TPNS if needed.
